src/js/components/TableCell/TableCell.js

```diff
@@ -33,66 +34,106 @@ const TableCell = forwardRef(
     ref,
   ) => {
     const theme = useContext(ThemeContext) || defaultProps.theme;
-    return (
-      <TableContext.Consumer>
-        {tableContext => {
-          let tableContextTheme;
-          if (tableContext === 'header') {
-            tableContextTheme = theme.table && theme.table.header;
-          } else if (tableContext === 'footer') {
-            tableContextTheme = theme.table && theme.table.footer;
-          } else {
-            tableContextTheme = theme.table && theme.table.body;
+    const tableContext = useContext(TableContext);
+    const cellRef = useForwardedRef(ref);
+    const containerRef = useRef();
+
+    // if window resizes, recalculate cell height so that content
+    // will continue to fill the height if the dimensions of the cell
+    // have changed
+    useEffect(() => {
+      const updateHeight = () => {
+        if (plain === 'noPad') {
+          const cell = cellRef.current;
+          const container = containerRef.current;
+          if (cell && container) {
+            container.style.height = '';
+            const cellRect = cell.getBoundingClientRect();
+
+            // height must match cell height otherwise table will apply some
+            // margin around the cell content
+            container.style.height = `${Math.max(
+              cellRect.height -
+                (border || theme.table[tableContext].border
+                  ? theme.global.borderSize.xsmall.replace('px', '')
+                  : 0),
+              0,
+            )}px`;
           }
-          // merge tabelContextTheme and rest
-          const mergedProps = { ...tableContextTheme, ...rest };
-          Object.keys(mergedProps).forEach(key => {
-            if (rest[key] === undefined)
-              mergedProps[key] = tableContextTheme[key];
-          });
-          // split out background, border, and pad
-          const cellProps = {
-            align: align || mergedProps.align || undefined,
-            background: background || mergedProps.background || undefined,
-            border: border || mergedProps.border || undefined,
-            pad: pad || mergedProps.pad || undefined,
-            verticalAlign:
-              verticalAlign || mergedProps.verticalAlign || undefined,
-          };
-          delete mergedProps.align;
-          delete mergedProps.background;
-          delete mergedProps.border;
-          delete mergedProps.pad;
-          delete mergedProps.verticalAlign;
+        }
+      };
+
+      window.addEventListener('resize', updateHeight);
+      updateHeight();
+      return () => {
+        window.removeEventListener('resize', updateHeight);
+      };
+    });
```

Comment:
Should this effect be rebuilt it the value for `plain` changes?

Reply:
Thanks yes I was missing the dependency array.